week4_2/src/pages/signup.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import apiClient from "../api/axios";
+import Contentbox from "../components/contentBox";
+import Title from "../components/Title";
+import InputForm from "../components/inputForm";
+import Cta from "../components/cta";
+
+const Signup = () => {
+  const [username, setUsername] = useState("");
+  const [nickname, setNickname] = useState("");
+  const [password, setPassword] = useState("");
+  const [passwordCheck, setPasswordCheck] = useState("");
+  const [isExist, setIsExist] = useState("");
```

Comment:
state 선언이 너무 많고, 밑에 props로 넘겨줄 친구가 너무 많아진 문제를 동시에 해결할 수 있는 방법으로
```suggestion
const initInputInfo = {
    username: "",
    nickname: "",
    password: "",
    passwordCheck: "",
    isExist: "",
    }
 const [inputInfo, setInputInfo] = useState(initInputinfo);
```
이런식으로 객체 하나로 state를 관리할 수 있다는거! 그럼 각각이 변경되었을 때의 로직도 하나로 묶을 수 있게된답니당